python/mxnet/profiler.py

```diff
@@ -54,5 +60,280 @@ def profiler_set_state(state='stop'):
 
 def dump_profile():
     """Dump profile and stop profiler. Use this to save profile
-    in advance in case your program cannot exit normally."""
+    in advance in case your program cannot exit normally.
+    """
     check_call(_LIB.MXDumpProfile())
+
+def dump_aggregate_stats(reset=False):
+    """Dump profile aggregate stats to console.
+    """
+    do_reset = 1 if reset is True else 0
+    check_call(_LIB.MXDumpAggregateProfileStats(int(do_reset)))
+
+def profiler_pause():
+    check_call(_LIB.MXProfilePause(int(1)))
+
+def profiler_resume():
+    check_call(_LIB.MXProfilePause(int(0)))
+
+class Domain(object):
+    """Profiling domain, used to group sub-objects like tasks, counters, etc into categories
+    Serves as part of 'categories' for chrome://tracing
+    Note: Domain handles are never destroyed.
+    """
+    def __init__(self, name):
+        """Profiling Domain class constructor
+            Parameters
+            ----------
+            name : string
+                Name of the domain
+        """
+        self.name = name
+        self.handle = ProfileHandle()
+        check_call(_LIB.MXProfileCreateDomain(c_str(self.name), ctypes.byref(self.handle)))
+
+    def __str__(self):
+        return self.name
+
+    def new_task(self, name):
+        """Create new Task object owned by this domain
+            Parameters
+            ----------
+            name : string
+                Name of the task
+        """
+        return Task(self, name)
+
+    def new_frame(self, name):
+        """Create new Frame object owned by this domain
+            Parameters
+            ----------
+            name : string
+                Name of the frame
+        """
+        return Frame(self, name)
+
+    def new_counter(self, name, value=None):
+        """Create new Counter object owned by this domain
+            Parameters
+            ----------
+            name : string
+                Name of the counter
+        """
+        return Counter(self, name, value)
+
+    def new_marker(self, name):
+        """Create new Marker object owned by this domain
+            Parameters
+            ----------
+            name : string
+                Name of the marker
+        """
+        return Marker(self, name)
+
+class Task(object):
+    """Profiling Task class
+    A task is a logical unit of work performed by a particular thread.
+    Tasks can nest; thus, tasks typically correspond to functions, scopes, or a case block
+    in a switch statement.
+    You can use the Task API to assign tasks to threads.
```

Comment:
1. Add standard Parameter section.
2. Add a note explaining the difference with Frame.

Reply:
I don't understand. There's a Parameter section in  the "def __init__(self, domain, name):" call.

Reply:
Should the section under __init__ be under the class name instead?

Reply:
I don;t see that elsewhere, i.e.

class Executor(object):
    """Executor is the object providing efficient symbolic graph execution and optimization.

    Examples
    --------
    >>> # typical approach to create an executor is to bind symbol
    >>> a = mx.sym.Variable('a')
    >>> b = mx.sym.Variable('b')
    >>> c = 2 * a + b
    >>> texec = c.bind(mx.cpu(), {'a': mx.nd.array([1,2]), 'b':mx.nd.array([2,3])})
    """
    def __init__(self, handle, symbol, ctx, grad_req, group2ctx):
        """Constructor, used Symbol.bind and Symbol.simple_bind instead.

        Parameters
        ----------
        handle: ExecutorHandle
            ExecutorHandle generated by calling `bind`.

        See Also
        --------
        Symbol.bind : to create executor.
        """
        if not isinstance(handle, ExecutorHandle):
            raise TypeError("Handle type error")

Reply:
I don't understand this comment :(